jupyterlab_bokeh_server/server.py

```diff
@@ -141,12 +144,177 @@ def cb():
 
     doc.add_periodic_callback(cb, 200)
 
+try:
+    from pynvml.nvml import *
+
+    nvml_init()
+    ngpus = nvmlDeviceGetCount()
+    gpu_handles = [ nvmlDeviceGetHandleByIndex(i) for i in range(ngpus) ]
+
+    def gpu(doc):
+        fig = figure(title="GPU Usage", sizing_mode="stretch_both", y_range=[0, 100])
+
+        gpu = [ nvmlDeviceGetUtilizationRates( gpu_handles[i] ).gpu+1 for i in range(ngpus) ]
+        left = list(range(len(gpu)))
+        right = [l + 0.8 for l in left]
+        source = ColumnDataSource({"left": left, "right": right, "gpu": gpu})
+        #colors = ["#000003", "#410967", "#932567", "#DC5039"] #, "#FBA40A", "#FCFEA4"]
+        colors = ['#440154', '#404387', '#29788E', '#22A784', '#79D151']
+        mapper = LinearColorMapper(palette=colors, low=0, high=100)
+
+        fig.quad(
+            source=source, left="left", right="right", bottom=0, top="gpu", color={"field": "gpu", "transform": mapper}
+        )
+
+        doc.title = "GPU Utilization [%]"
+        doc.add_root(fig)
+
+        def cb():
+            source.data.update({"gpu": [ nvmlDeviceGetUtilizationRates( gpu_handles[i] ).gpu+0.1 for i in range(ngpus) ]})
+
+        doc.add_periodic_callback(cb, 200)
 
-routes = {
-    "/cpu": cpu,
-    "/resources": resource_timeline,
-}
+    def pci(doc):
+
+        tx_fig = figure(title="TX Bytes [KB]", sizing_mode="stretch_both", y_range=[0, 10000])
```

Comment:
Is this kB/s or kB?  If kB/s, why a maximum of 10000, this seems small?

I would expect something closer to MB/s as a unit and for it to range up to several GB/s for a maximum.